src/nrnoc/treeset.cpp

```diff
@@ -1991,7 +2008,7 @@ static void nrn_matrix_node_alloc(void) {
             v_setup_vectors();
             return;
         } else {
-            if (nt->_actual_rhs != (double*) 0) {
+            if (nt->node_rhs_storage() != nullptr) {
```

Comment:
I'm a bit suspicious here, I **think** that `node_rhs_storage()` will not normally ever return `nullptr` (maybe for the zeroth thread with zero Nodes in the application). So this check may not have the same meaning that it did before.

Reply:
I think you're right :+1: I'm wondering if we need to check whether some other vector (like `actual_d`) was already allocated though to avoid reallocating everything

Reply:
OK, we can revisit that when we also move `d` to use the new data structures.